chaging folder structure as per target platform for windows plans for…

raviduddela opened the pull request:
… first ten pkgs

Signed-off-by: RaviDuddela <rduddela@progress.com>

mwrock commented:
You don't need to create a windows platform folder for plan.ps1 because there is no other platform we support that can build a plan.ps1. The plan-build scripts will use <root>/plan.ps1 to build a windows plan.

raviduddela commented:
> You don't need to create a windows platform folder for plan.ps1 because there is no other platform we support that can build a plan.ps1. The plan-build scripts will use /plan.ps1 to build a windows plan.

Hi matt, as we are moving base plans to core plans there are pkgs which needs to build for diff platforms linux, windows and linux2, so we decided to keep the default root folder for linux files and other plan files in a platform specific folder.I have changed ci script and added target in bldrtoml but at a time its running only linux plan. Is this the expected behaviour. can't we run builds for two platforms simultaneously.

mwrock commented:
yes you can have builds that target multiple platforms for a single package. You want to make sure that the plan file in the settings UI on bulder is correctly matched for each platform. One problem I see with your approach is that you will need to duplicate linux and linux2 plans. In the vast majority of cases, a linux plan.sh and a linux2 plan.sh file does not need to differ at all. So you can just put the file on the root and both targets will use the same file. It is only when you need different code in the linux2 plan where you need a separate platform folder.

raviduddela commented:
> yes you can have builds that target multiple platforms for a single package. You want to make sure that the plan file in the settings UI on bulder is correctly matched for each platform. One problem I see with your approach is that you will need to duplicate linux and linux2 plans. In the vast majority of cases, a linux plan.sh and a linux2 plan.sh file does not need to differ at all. So you can just put the file on the root and both targets will use the same file. It is only when you need different code in the linux2 plan where you need a separate platform folder.

Yes matt. plan file path we are changing in the builder UI. Yes for most of the linux2 plans the plan file is same as linux except for 8-9 pkgs. we are not duplicating for linux2.seperate platform folder is present for that 9 pkgs and we changed in the builder too. this is for windows platform. Ex. python has for windows, linux and linux2 . we have for some 20+ pkgs which has both linux and windows plans. so planning to create windows platform folder and change in builder too. but the build kite build is not happening . is this works or we need to keep both plan.sh and plan.ps1 and corresponding patch files in the root folder? the test scripts also need to put under tests dir for both windows and linux. however the Buildkite builds were also not happening earlier for windows.

mwrock commented:
My recomendation would be to keep the plan.ps1 files in the root because adding the target folder will have no benefit and just create an additional folder layer. Buildkite/expeditor do not build the packages for core-plans. Well...they do in the verify pipeline but those artifacys never get uploaded to builder. After merge, builder handles the build that gets uploaded in builder. In order for builder to do that propperly 2 things need to happen:

1. The builder settings need to be corrected for the linux plans moving over from base-plans for the plan file connection.
2. We need to re-enable auto builds for the core origin on builder

raviduddela commented:
> My recomendation would be to keep the plan.ps1 files in the root because adding the target folder will have no benefit and just create an additional folder layer. Buildkite/expeditor do not build the packages for core-plans. Well...they do in the verify pipeline but those artifacys never get uploaded to builder. After merge, builder handles the build that gets uploaded in builder. In order for builder to do that propperly 2 things need to happen:
> 
> 1. The builder settings need to be corrected for the linux plans moving over from base-plans for the plan file connection.
> 2. We need to re-enable auto builds for the core origin on builder

Hi matt,
1. we have moved base-plans to core and we changed builder settings to new location for all the linux plans.
2. some packages have both plan files .sh and .ps1 . so when we move to core plans they end up in one root folder.
its fine if we have only plan files but there are additional files and folders for some packages like patches and config folders , patches with same name and tests directory. if these exists in the same root directory there might be some confusion. so to make code more friendly and readable thought of maintaining code as per platform default root folder being for linux. ex: make,openssl, Nginx etc. 
3. we need to make corresponding changes in builder settings and bin/ci/build-and-run-test.ps1 to run pipeline for verification if buildkite works.
4. we can wait for reenabling of auto builds.

raviduddela commented:
Closing as we concluded to maintain both plans in the root package folder
